### ISSUE TITLE
about [RET] token

### ISSUE THREAD

**Issue by haokunwen** [dhg-wei/MCL#1]
Hi~ Great work! 
I am curious about how LLM determines when to generate [RET] tokens for image retrieval. Does the LLM generate [RET] tokens solely based on contextual information, or is there a specific mechanism designed to prompt the generation of [RET] tokens?

Thank you!

**Comment by dhg-wei** [comment]
Hi, thank you for your interest in our work. 

Following [FORMAGe](https://github.com/kohjingyu/fromage), we add the [RET] tokens to the LLM tokenizer, which can be regarded as natural words.

For specific image retrieval tasks, we structure the input of LLM as follows:  {<Multimodal context tokens~(e.g., image tokens and caption tokens)>, <Prompt tokens~(e.g., "the target image is")>,  <[RET] Tokens>}.

In more open-ended scenarios, such as multimodal dialog, the [RET] tokens can be predicted as the next word based on the context. Mechanisms like logits-scale can be used to adjust the probability of the generating [RET] tokens.

**Comment by haokunwen** [comment]
Got it! Thank you for the detailed response.
